Added a ActivityState and MergeTargetId into the [Customer] entity. (#345)
MewsSystems/gitbook-connector-api: operations/customers.md

| `ItalianDestinationCode` | string | optional | Value of Italian destination code. |
| `ItalianFiscalCode` | string | optional | Value of Italian fiscal code. |
| `CompanyId` | string | optional | Unique identifier of [Company](companies.md#company) the customer is associated with. |

> | `MergeTargetId` | string | optional | `Id` of the account (Customer/Company) where the customer is linked to. |

Reviewer: Two small suggestions:
1. Grammar fix - better to say:
| `MergeTargetId` | string | optional | `Id` of the account (customer/company) to which this customer is linked. |

2. Can we include a link to 'account', does it exist as an entity in the API? Or is it some sort of co-product for Company and Customer? But would it always be Customer in this case, because the name "MergeTargetId" makes me think that this is used when merging two customer profiles? I don't know, but I just don't want it to be confusing for the user. Thanks.

Author: word fixed. The account doesn't exist, it is an entity in our systems - it is Customer / Company.. but maybe can be in the future something different. We can remove that word account for clarity, or keep it as it is with these examples..

Reviewer: Thanks @pindurav. Just to be clear, are you saying that there is no such entity or object as account, it's just a word used to mean either Company or Customer? That's ok, but my next question then is how does the user know if `MergeTargetId` points to a Customer ID or a Company ID?

> | `MergeTargetId` | string | optional | `Id` of the account ([Customer](#customer)/[Company](companies.md#company)) to which this customer is linked. |

Reviewer 2: ```suggestion
| `MergeTargetId` | string | optional | Unique identifier of the account ([Customer](#customer)) to which this customer is linked. |
```

| `ActivityState` | string | optional | [State](#activity-state) of returned customer. |

> 

Reviewer: Suggest new wording:
| `ActivityState` | string | optional | [Activity State](#activity-state) of customer record, i.e. whether active or deleted. |

> | `ActivityState` | string | optional | [Activity State](#activity-state) of customer record, i.e. whether active or deleted. |

Reviewer 2: ```suggestion
| `ActivityState` | string | required | [Activity State](#activity-state) of customer record, i.e. whether active or deleted. |
```

> | `ActivityState` | string | required | [Activity State](#activity-state) of customer record, i.e. whether active or deleted. |
> 

Reviewer 2: ```suggestion
| `ActivityState` | string | required | [Activity State](#activity-state) of customer record, i.e. whether active or deleted. |
```

* `SendMarketingEmails`
* ...

> #### Activity State

Reviewer 2: The activity state is already defined in Vouchers object, might be worth to extract it to separate page wit h such a common elements.

Reviewer 3: What might be the best page for that? just reference it from vouchers? I think if we don't have such a page yet typically we have duplicated objects... same for update values, etc.

Reviewer: Current policy on this question is as follows:
1. Where there are common data objects referenced in multiple places, define the object once close to where it is first used or most frequently used and link to that one instance, even if on a separate page
2. EXCEPT... for very minor data objects (like Activity State or Update Value), these can be duplicated on all the pages where they are referenced
3. Regardless, a data object should never be defined more than once on the same page; and if multiple definitions have the same name, then they must have the same definition!
4. In general, being consistent is more important than the actual method used

Of course the policy can be changed, but that is the current policy. So in this case, it's fine to define Activity State on this page as long as (1) it is only defined once on the page; (2) the definition is exactly the same as appears in vouchers.md; and (3) for consistency, all instances of references to Activity State on this page should link to the same definition (i.e. #activity-state).

Phew!

> * `Deleted`
> * `Active`

Reviewer 2: ```suggestion
* `Active`
* `Deleted`
```
